app/react/Thesauri/actions/thesauriActions.js

```diff
@@ -19,27 +18,24 @@ export function saveThesaurus(thesaurus) {
 }
 
 export function importThesaurus(thesaurus, file) {
-  return dispatch =>
-    new Promise(resolve =>
-      superagent
-        .post(`${APIURL}thesauris`)
-        .set('Accept', 'application/json')
-        .set('X-Requested-With', 'XMLHttpRequest')
-        .field('thesauri', JSON.stringify(thesaurus))
-        .attach('file', file, file.name)
-        .on('response', response => {
-          const data = JSON.parse(response.text);
-          if (response.status === 200) {
-            dispatch({ type: types.THESAURI_SAVED });
-            notifications.notify(t('System', 'Data imported', null, false), 'success')(dispatch);
-            dispatch(formActions.change('thesauri.data', data));
-          } else {
-            notifications.notify(t('System', data.error, null, false), 'danger')(dispatch);
-          }
-          resolve();
-        })
-        .end()
-    );
+  return async dispatch => {
+    try {
+      const headers = {
+        Accept: 'application/json',
+        'X-Request-With': 'XMLHttpRequest',
+      };
```

Comment:
Thesaurus import is not working, please take a look

Reply:
Worked on this